File: C4_InnerTube/InnerTubeCS/SharedUtilities/ZuneSync.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Linq;
using System.Text;
using Microsoft.Win32;
using System.Collections.ObjectModel;
using System.IO;

namespace SharedUtilities
{

    public enum ZuneMonitoredFolders
    {
        MonitoredAudioFolders,
        MonitoredVideoFolders,
        MonitoredPhotoFolders,
        MonitoredPodcastFolders
    }

    public class ZuneSync : IVideoService
    {
        
        public static string[] GetZuneMonitoredFolders(ZuneMonitoredFolders folder)
        { 
            string hive = @"Software\Microsoft\Zune\Groveler\";
            string[] values = (string[])Registry.CurrentUser.OpenSubKey(hive).GetValue(folder.ToString());
            return values;
        }


        #region VideoService Members
        public void Sync(string filePath)
        {
            string[] currentFolders = ZuneSync.GetZuneMonitoredFolders(ZuneMonitoredFolders.MonitoredVideoFolders);

            bool found = currentFolders.Contains(filePath);
            //check if we are already added the files to the folder
            if (!found)
            {
                //copy the files to the first specified directory   
                if (currentFolders.Length > 0)
                {
                    string destinationPath = currentFolders[0];
                    string[] Files = Directory.GetFiles(filePath, "*.wmv", SearchOption.TopDirectoryOnly);
                    foreach (var f in Files)
                    {
                        File.Copy(f, destinationPath, true);
                    }

                }
                else
                {
                    throw new ArgumentException("Zune is not configured to monitor *any* " +
                    "folders, to fix this, open zune.exe, click settings, and add a video folder");
                }                                
            }



        }


        #endregion

    }


}

File: C04_InnerTube/InnerTubeVB/SharedUtilities/ZuneSync.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Collections
Imports System.Linq
Imports System.Text
Imports Microsoft.Win32
Imports System.Collections.ObjectModel
Imports System.IO

Namespace SharedUtilities

	Public Enum ZuneMonitoredFolders
		MonitoredAudioFolders
		MonitoredVideoFolders
		MonitoredPhotoFolders
		MonitoredPodcastFolders
	End Enum

	Public Class ZuneSync
		Implements IVideoService
		Public Shared Function GetZuneMonitoredFolders(ByVal folder As ZuneMonitoredFolders) As String()
			'Pull registry
            Dim hive As String = "Software\Microsoft\Zune\Groveler\"
            Dim values() As String = CType(My.Computer.Registry.CurrentUser.OpenSubKey(hive).GetValue(folder.ToString()), String())

			Return values
		End Function


		#Region "VideoService Members"
		Public Sub Sync(ByVal filePath As String) Implements IVideoService.Sync
			Dim currentFolders() As String = ZuneSync.GetZuneMonitoredFolders(ZuneMonitoredFolders.MonitoredVideoFolders)

			Dim found As Boolean = currentFolders.Contains(filePath)
			'check if we are already adding the files to the folder
			If (Not found) Then

				'copy the files to the first specified directory   
				If currentFolders.Length >0 Then
					Dim destinationPath As String = currentFolders(0)
					Dim Files() As String = Directory.GetFiles(filePath, "*.wmv", SearchOption.TopDirectoryOnly)
					For Each f In Files
						File.Copy(f, destinationPath, True)
					Next f

				Else
					Throw New ArgumentException("Zune is not configured to monitor *any* " & "folders, to fix this, open zune.exe, click settings, and add a video folder")
				End If

			End If
        End Sub
		#End Region

	End Class


End Namespace



